bin/propolis-standalone/src/main.rs

```diff
@@ -1277,20 +1152,43 @@ fn setup_instance(
     });
 
     // generate SMBIOS data and expose via fw_cfg
-    let smbios::TableBytes { entry_point, structure_table } =
-        generate_smbios(SmbiosParams {
+    let smbios::TableBytes { entry_point, structure_table } = {
+        let cpuid_ident = cpuid_ident
+            .unwrap_or_else(|| cpuid::host_query(cpuid::Ident(0x1, None)));
+        let cpuid_procname = cpuid_procname.unwrap_or_else(|| {
+            if cpuid::host_query(cpuid::Ident(0x8000_0000, None)).eax
+                >= 0x8000_0004
```

Comment:
boring but funny note: the oldest CPUs that don't support cpuid leaves 0x8000_000[1..4] are, i think, Pentium II and K5. so as long as Propolis is running on a CPU newer than those, this arm is always taken?